src/sparse/impl/KokkosSparse_spgemm_impl_symbolic.hpp

```diff
@@ -229,21 +229,23 @@ struct KokkosSPGEMM
 
     const nnz_lno_t team_row_begin = teamMember.league_rank() * team_row_chunk_size;
     const nnz_lno_t team_row_end = KOKKOSKERNELS_MACRO_MIN(team_row_begin + team_row_chunk_size, numrows);
-
+    nnz_lno_t chunk_size = 0;
 
     //get memory from memory pool.
     volatile nnz_lno_t * tmp = NULL;
     size_t tid = get_thread_id(team_row_begin + teamMember.team_rank());
     while (tmp == NULL){
       tmp = (volatile nnz_lno_t * )( m_space.allocate_chunk(tid));
+      // issue-508, TODO: chunk_size = ??
     }
 
     //set first to globally used hash indices.
     nnz_lno_t *globally_used_hash_indices = (nnz_lno_t *) tmp;
     tmp += pow2_hash_size;
 
     //create hashmap accumulator.
-    KokkosKernels::Experimental::HashmapAccumulator<nnz_lno_t,nnz_lno_t,nnz_lno_t> hm2;
+    KokkosKernels::Experimental::HashmapAccumulator<nnz_lno_t,nnz_lno_t,nnz_lno_t,KokkosKernels::Experimental::HashOpType::bitwiseAnd> 
+    hm2(chunk_size, pow2_hash_func, nullptr, nullptr, nullptr, nullptr);
```

Comment:
Does this work now, with chunk_size = 0? I'm worried that the new insert routines will compare against that and return __insert_full immediately.

Reply:
For the first arg, you should just be able to pass INT_MAX (or the equivalent for whatever type) because the chunk size is computed to always be big enough for the longest row, so insertion should never return __insert_full.

Reply:
Good catch! `chunk_size` should be `MaxRoughNonZero` (see https://github.com/kokkos/kokkos-kernels/pull/731/files/ebab3bc86c0e9aebf28b0cc3cdaaaa96419bf058#diff-bba5fd8c6962fac594fa0e5918d87e88L261)!

Reply:
@e10harvey Good, that is a better value to use than INT_MAX. So did this not get caught by sparse unit tests?

Reply:
It did not get caught by the sparse unit tests. I'll track that down and update the tests.

Reply:
@brian-kelley: I moved this to a separate issue: https://github.com/kokkos/kokkos-kernels/issues/740.